packages/typespec-client-generator-core/src/package.ts

```diff
@@ -755,7 +731,7 @@ function getSdkEndpointParameter<TServiceOperation extends SdkServiceOperation =
       variantTypes: types,
       name: createGeneratedName(context, rawClient.service, "Endpoint"),
       isGeneratedName: true,
-      crossLanguageDefinitionId: getCrossLanguageDefinitionId(context, rawClient.service),
+      crossLanguageDefinitionId: `${getCrossLanguageDefinitionId(context, rawClient.service)}.Endpoint`,
```

Comment:
nit, in the test case, the value is `Contoso.WidgetManager.endpoint`

Reply:
for the parameter, it is `endpoint`, for the union type, it is `Endpoint`.